### PR TITLE
Fix transformer pruning example

### PR DESCRIPTION
**Pull request by xiaowu0162**
Fixes:
- config list indentation in example and docs
- example formatting 
- introduce example in the docs

### CODE REVIEW

**Comment by xiaowu0162** [comment]
@zheng-ningxin @J-shang

**Review by J-shang — APPROVED** [review]

File: examples/model_compress/pruning/transformers/transformer_pruning.py [hunk 1]
```diff
@@ -316,13 +316,11 @@ def forward_runner(model):
...
         "op_types": ["Linear"],
         "op_names": [x for layer in attention_name_groups[:6] for x in layer]
-    },
-        {
-            "sparsity": args.sparsity / 2,
-            "op_types": ["Linear"],
-            "op_names": [x for layer in attention_name_groups[6:] for x in layer]
-        }
-    ]
+    }, {
```
Comment (zheng-ningxin) [inline examples/model_compress/pruning/transformers/transformer_pruning.py:319]: https://nni.readthedocs.io/en/latest/Compression/Pruner.html#transformer-head-pruner 
I mean the example in the documentation here has the wrong indent

Comment (xiaowu0162) [reply examples/model_compress/pruning/transformers/transformer_pruning.py:319]: fixed

**Comment by xiaowu0162** [comment]
will add fixes for some other issues related to the example

**Review by zheng-ningxin — APPROVED** [review]

**Review by QuanluZhang — APPROVED** [review]

**Pull request merged**